internal/venice-common/src/main/java/com/linkedin/venice/writer/VeniceWriterFactory.java

```diff
@@ -41,6 +43,7 @@ public <K, V, U> VeniceWriter<K, V, U> createVeniceWriter(VeniceWriterOptions op
     return new VeniceWriter<>(
         options,
         props,
-        producerAdapterFactory.create(props, options.getTopicName(), options.getBrokerAddress()));
+        Lazy.of(() -> producerAdapterFactory.create(props, options.getTopicName(), options.getBrokerAddress())),
```

Comment:
I'm curious about this... have you found specific code paths that were instantiating a `VW` but then not actually using it to produce anything?

In many places, we are already wrapping the `VW` into `Lazy`, or else use other patterns of lazy init (e.g. `computeIfAbsent`). So now we'll be doing double-lazy wrapping in many places. In theory it's fine but it seems a bit dirty to me... maybe the clean thing to do is to remove the "outer" lazy wrapping and just keep this inner one, or perhaps to make the remaining unwrapped `VW` into `Lazy<VW>`. I don't have a strong opinion either way but I feel like it's something we should consider.

Anyway, I don't mean to say you should embark on a big refactoring in this PR, but I am interested in hearing your opinion on the above... I imagine you looked at this code more recently than I did if you're doing this, so maybe you caught some interesting context IDK about along the way.

Reply:
You're right. I just double-checked the code, and, except for two place (and none of them are in server startup path), we're indeed initializing `VW` only when we need it. One trigger to add lazy initialization for VW's producer was based on hazy log reading, where I saw producers being one after another during startup . I should have properly checked the code before adding this code.

Will update the PR to remove lazy init. Thanks!